### PR TITLE
✨ Add e2e release tests for cron job

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Adds e2e release tests for cron job.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: cron/controller/Dockerfile [hunk 1]
```diff
@@ -27,6 +26,7 @@ ARG TARGETARCH
 RUN CGO_ENABLED=0 make build-pubsub
 
 FROM gcr.io/distroless/base:nonroot@sha256:bc84925113289d139a9ef2f309f0dd7ac46ea7b786f172ba9084ffdb4cbd9490
+USER 0
```
Comment (oliverchang) [inline cron/controller/Dockerfile:29]: what's this needed for?

**Review by oliverchang — APPROVED** [review]
nice! what's the expected workflow for determining if the e2e passed and we can promote this to prod?

File: cron/controller/Dockerfile (hunk 1 above)
Comment (azeemshaikh38) [reply cron/controller/Dockerfile:29]: So I'm running into a weird issue where Go is unable to open "cron/data/projects.csv" file due to permission error. According to Dockerfile reference, the default user for the `COPY` command is `0`. So I tried setting `USER` to `0` before `ENTRYPOINT` and that seems to solve this. Not sure if this is the right solution though. Any ideas?

Comment (oliverchang) [reply cron/controller/Dockerfile:29]: That's weird. How did it work before?

It does seem a bit hacky to set the user to root given that we're using a "nonroot" image. Can you try using another distroless image (not the "nonroot" one) and see if that fixes things?

Comment (azeemshaikh38) [reply cron/controller/Dockerfile:29]: So this issue happened due to https://github.com/ossf/scorecard/pull/706. I only caught it now when writing these e2e tests. Before that we were embedding the file and not reading it through `os.Open`.

Will try the "nonroot" image and get back to you.

Comment (azeemshaikh38) [reply cron/controller/Dockerfile]: Fixed it, thanks to @dlorenc. The permission of the file in the repo only allowed `root` to have read access. Updated that now.

File: cron/releasetest/k8s/controller.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  # Daily at midnight PST.
+  schedule: "0 7 * * *"
+  concurrencyPolicy: "Forbid"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: controller
+            image: gcr.io/openssf/scorecard-batch-controller-releasetest:latest
```
Comment (laurentsimon) [inline cron/releasetest/k8s/controller.yaml:29]: is it possible to pin this? Does dependabot support updating it?
I have a tracking issue to add support for this in our Frozen-Deps check.

File: cron/worker/main.go [hunk 3]
```diff
@@ -127,7 +127,7 @@ func createNetClients(ctx context.Context) (
 func startMetricsExporter() (monitoring.Exporter, error) {
 	exporter, err := monitoring.GetExporter()
 	if err != nil {
-		return nil, fmt.Errorf("error during NewStackDriverExporter: %w", err)
+		return nil, fmt.Errorf("error during monitoring.GetExporter: %w", err)
```
Comment (laurentsimon) [inline cron/worker/main.go:130]: shall we start using the new command to create errors?

**Comment by azeemshaikh38** [comment]
> nice! what's the expected workflow for determining if the e2e test passed and we can promote this to prod?

Right now, its manual. This is mostly for sanity checking that the cron job doesn't fail for some reason.

I still need to figure out the process for prod releases. Will discuss this in the meeting.

File: cron/worker/main.go (hunk 3 above)
Comment (azeemshaikh38) [reply cron/worker/main.go:130]: Yes, we should. Forgot about it in this PR.

File: cron/releasetest/k8s/controller.yaml (hunk 2 above)
Comment (azeemshaikh38) [reply cron/releasetest/k8s/controller.yaml:29]: Hmm, the idea is that this should always pull the most recent image to test. Pinning doesn't really make sense here.

**Pull request merged**